src/main/scala/io/iohk/ethereum/network/discovery/PeerDiscoveryManager.scala

```diff
@@ -6,25 +6,49 @@ import akka.util.ByteString
 import cats.effect.Resource
 import io.iohk.ethereum.db.storage.KnownNodesStorage
 import io.iohk.scalanet.discovery.ethereum.v4
+import io.iohk.scalanet.discovery.ethereum.{Node => ENode}
 import monix.eval.Task
-import monix.execution.Scheduler
+import monix.execution.{Scheduler, BufferCapacity}
+import monix.tail.Iterant
+import monix.catnap.ConsumerF
 import scala.util.{Failure, Success}
+import scala.util.Random
 
 class PeerDiscoveryManager(
     localNodeId: ByteString,
     discoveryConfig: DiscoveryConfig,
     knownNodesStorage: KnownNodesStorage,
     // The manager only starts the DiscoveryService if discovery is enabled.
-    discoveryServiceResource: Resource[Task, v4.DiscoveryService]
+    discoveryServiceResource: Resource[Task, v4.DiscoveryService],
+    randomNodeBufferSize: Int
 )(implicit scheduler: Scheduler)
     extends Actor
     with ActorLogging {
 
+  // Create a random nodes iterator on top of the service so the node can quickly ramp up its peers.
+  val discoveryResources = for {
+    service <- discoveryServiceResource
+
+    randomNodes = Iterant
+      .repeatEvalF {
+        Task(log.debug("Pulling random nodes on demand...")) >>
+          service.lookupRandom
```

Comment:
This is someting i just noticed, that lookupRandom uses `sigalg.newKeyPair._1` to generate random key for each lookup and our implementation  `Secp256k1SigAlg` acutally generates real keys which can be pretty costly performance wise. Maybe for random lookups we should just generate random 64 bytes values ?

Reply:
We can do that, although I don't think it's going to have any noticeable impact. On my machine generating 1000 new keys takes ~650ms. Generating 1000 random 64 byte arrays is just 4ms, but during my testing I only saw 1 lookup happening every ~10 seconds.

Reply:
Done, although I preferred the simplicity of the original.